scripts/affiliate_server.py

```diff
@@ -77,12 +86,73 @@ def get_current_amazon_batch() -> Batch:
     return batch
 
 
+def get_isbns_from_book(book: dict) -> list[str]:  # Singular: book
+    return [str(isbn) for isbn in book.get('isbn_10', []) + book.get('isbn_13', [])]
+
+
+def get_isbns_from_books(books: list[dict]) -> list[str]:  # Plural: books
+    return sorted(set(itertools.chain(*[get_isbns_from_book(book) for book in books])))
+
+
+def is_book_needed(book: dict, edition: dict) -> bool:
+    """
+    Should an OL edition's metadata be updated with Amazon book data?
+
+    :param book: dict from openlibrary.core.vendors.clean_amazon_metadata_for_load()
+    :param edition: dict from web.ctx.site.get_many(edition_ids)
+    """
+    needed_book_fields = {}  # dict of book fields that should be copied to the edition
+    for book_field, edition_field in AZ_OL_MAP.items():
+        if field_value := book.get(book_field) and not edition.get(edition_field):
+            needed_book_fields[book_field] = field_value
+    if needed_book_fields:  # Log book fields that should to be copied to the edition
+        fields = ", ".join(f"{k}={v}" for k, v in needed_book_fields.items())
+        logger.debug(f"{edition['key']} needs {fields}")
+    return bool(needed_book_fields)
```

Comment:
Since we're not doing anything with `needed_book_fields` other than essentially checking bool / length, we could simplify the code logic to just keep track of fields from the `edition` which are missing (the amazon fields do not add more info + would shorten logs)

Not blocking

Reply:
We made this into a list to deal with a non-existent ol edition.